Storage/Processor/Processor.php

```diff
@@ -259,4 +248,72 @@ public function getResource(Config $config, string $filename, bool $noCache = fa
         $this->saveGeneratedResourceToCache($generatedResource, $config, $filename);
         return $generatedResource;
     }
+
+    private function getResponseFromStreamInterface(StreamInterface $streamInterface, Request $request): StreamedResponse
+    {
+        $response = new StreamedResponse(function () use ($streamInterface) {
+            if ($streamInterface->isSeekable()) {
+                $streamInterface->rewind();
+            }
+
+            while (!$streamInterface->eof()) {
+                echo $streamInterface->read(self::BUFFER_SIZE);
+            }
+            $streamInterface->close();
+        });
+
+        if (null === $fileSize = $streamInterface->getSize()) {
+            return $response;
+        }
+        $response->headers->set('Content-Length', strval($fileSize));
+
+        if ($streamInterface->isSeekable()) {
+            $response->headers->set('Accept-Ranges', $request->isMethodSafe() ? 'bytes' : 'none');
+        }
+
+        $range = $request->headers->get('Range');
+
+        if ($range === null) {
+            return $response;
+        }
+
+        list($start, $end) = explode('-', substr($range, 6), 2) + [0];
+
+        $end = ('' === $end) ? $fileSize - 1 : (int) $end;
+
+        if ('' === $start) {
+            $start = $fileSize - $end;
+            $end = $fileSize - 1;
+        } else {
+            $start = (int) $start;
+        }
+
+        if ($start > $end) {
+            return $response;
+        }
+
+        if ($start < 0 || $end > $fileSize - 1) {
```

Comment:
I think the start/end logic should be handled in a different object. So that the if test can be (dummy code) $object->meaningfullFuntionName() . That functionname should cover the load of your if test, because the current test is not very speaking for itself. 

I would interpret this if test `if ($start < 0 || $end > $fileSize - 1)` as if you try to read before the start of after the end of the file, but that does not make much sense in my head?

Reply:
Looking at the code in the if, the check makes sense, but the following if would be less tough to read:

`if ($streamRange->isOutOfRange())`

Reply:
Refactored as suggested b681a9909166b190d18745755b2624f615a4982c